src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -821,6 +816,12 @@ def resolve_imports_and_packages(
         else session.get_session_stage()
     )
 
+    import_stage = (
```

Comment:
Let's define import_stage first then `upload_stage = import_stage if is_permanent else session_stage`.

Reply:
I wonder if there's a better name for `upload_stage`, because technically we need to **upload** files to `import_stage` too. How about something like `sproc_stage` vs `import_stage`? 
In addition, could you please add a one line documentation for these args to clarify?

Reply:
I agree with your point but the stage is used for udfs as well. How about `import_only_stage` and `upload_and_import_stage`?

Reply:
I don't think we upload files to import stage, but it is the other way around (we import the upload stage). And as Afroz mentioned, the import/upload difference is application to UDF families as well. I cannot come up with a better name than what Afroz suggested.

Reply:
done